internal/handlers/iso_test.go

```diff
@@ -136,6 +136,12 @@ var _ = Describe("ServeHTTP", func() {
 			})
 
 			It("returns a full image", func() {
+				assistedServer.SetHandler(0,
```

Comment:
Maybe remove the `AppendHandlers` call from `BeforeEach` and create a helper for expecting a call to the assisted server.
That should allow you to deduplicate all of this while also expecting the different iso types.